addons/purchase_mrp/models/purchase_mrp.py

```diff
@@ -1,19 +1,59 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import api, fields, models
-from odoo.tools import float_compare
+from odoo import api, fields, models, _
 
 
 class MrpProduction(models.Model):
     _inherit = 'mrp.production'
 
+    mto_purchase_order_count = fields.Integer("Count of generated PO", compute='_compute_mto_purchase_order_count', store=True,
+                                              groups='purchase.group_purchase_user')
+
+    @api.depends('procurement_group_id.stock_move_ids.created_purchase_line_id.order_id')
+    def _compute_mto_purchase_order_count(self):
+        for production in self:
+            production.mto_purchase_order_count = len(production.procurement_group_id.stock_move_ids.created_purchase_line_id.order_id)
+
     def _get_document_iterate_key(self, move_raw_id):
         iterate_key = super(MrpProduction, self)._get_document_iterate_key(move_raw_id)
         if not iterate_key and move_raw_id.created_purchase_line_id:
             iterate_key = 'created_purchase_line_id'
         return iterate_key
 
+    def action_view_mto_purchase_orders(self):
+        self.ensure_one()
+        return {
+            'name': _('Purchase Order generated from %s' % self.name),
+            'domain': [('id', 'in', self.procurement_group_id.stock_move_ids.created_purchase_line_id.order_id.ids)],
```

Comment:
https://github.com/odoo/odoo/pull/43366#issuecomment-575114872
The same

Reply:
It is legit because, it called from a form view